plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/CountingAccessFileHiveMetastore.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg;
+
+import io.trino.plugin.hive.HdfsEnvironment;
+import io.trino.plugin.hive.NodeVersion;
+import io.trino.plugin.hive.PartitionStatistics;
+import io.trino.plugin.hive.authentication.HiveIdentity;
+import io.trino.plugin.hive.metastore.Database;
+import io.trino.plugin.hive.metastore.MetastoreConfig;
+import io.trino.plugin.hive.metastore.Table;
+import io.trino.plugin.hive.metastore.file.FileHiveMetastore;
+import io.trino.plugin.hive.metastore.file.FileHiveMetastoreConfig;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class CountingAccessFileHiveMetastore
+        extends FileHiveMetastore
+{
+    public static final String GET_TABLE = "getTable";
+    public static final String GET_ALL_DATABASES = "getAllDatabases";
+    public static final String GET_DATABASE = "getDatabase";
+    public static final String GET_TABLE_WITH_PARAMETER = "getTablesWithParameter";
+    public static final String GET_TABLE_STATISTICS = "getTableStatistics";
+    private Map<String, Integer> methodInvocations;
+
+    public CountingAccessFileHiveMetastore(NodeVersion nodeVersion, HdfsEnvironment hdfsEnvironment, MetastoreConfig metastoreConfig, FileHiveMetastoreConfig config)
+    {
+        super(nodeVersion, hdfsEnvironment, metastoreConfig, config);
+        methodInvocations = new ConcurrentHashMap<>();
+        resetCounters();
+    }
+
+    public Map<String, Integer> getMethodInvocations()
+    {
+        return methodInvocations;
+    }
+
+    public void resetCounters()
+    {
+        methodInvocations.put(GET_TABLE, 0);
+        methodInvocations.put(GET_ALL_DATABASES, 0);
+        methodInvocations.put(GET_DATABASE, 0);
+        methodInvocations.put(GET_TABLE_WITH_PARAMETER, 0);
+        methodInvocations.put(GET_TABLE_STATISTICS, 0);
+    }
+
+    @Override
+    public synchronized Optional<Table> getTable(HiveIdentity identity, String databaseName, String tableName)
+    {
+        methodInvocations.compute(GET_TABLE, (k, v) -> v + 1);
```

Comment:
`merge(k, 0, Integer::add)` would be more canonical,
but anyway, let's use multiset.

Reply:
ok